certs/certs.go

```diff
@@ -276,5 +299,5 @@ func certsToRemove(oldCerts, newCerts []*x509.Certificate) map[string]*x509.Cert
 		}
 	}
 
-	return certsToRemove
+	return certsToRemove, nil
```

Comment:
Can't comment in the unchanged middle of this function, but with the changed behaviour such that we error if `len(newCerts) == 0` should we also check between the 2 loops that `newCertMap` isn't 0 length?